irispy/obsid.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from pkg_resources import resource_filename
+
+
+class Obsid(dict):
+    """A class to convert the IRIS OBS ID to human-readable format.
+
+    Parameters
+    ----------
+    obsid: `int`
+        IRIS OBS ID to query. Needs to be a valid OBS ID, with 10 digits.
+
+    Notes
+    -----
+    Currently, only OBS IDs with version numbers 36, 38, and 40 are
+    supported. Calibration OBS IDs (version 42) are not yet supported.
+
+    Examples
+    --------
+    Quickly show OBS ID parameters:
+
+    >>> obsid.Obsid(3675508564)
+    IRIS OBS ID 3675508564
+    ----------------------
+    Description:            Large dense 96-step raster 31.35x120 96s
+    SJI filters:                                    C II   Mg II w s
+    SJI field of view:                                       120x120
+    Exposure time:                                               8.0 s
+    Binning:                               Spatial x 1, Spectral x 1
+    FUV binning:                         FUV spectrally rebinned x 4
+    SJI cadence:                                 SJI cadence default
+    Compression:                                Lossless compression
+    Linelist:                                       Flare linelist 1
+
+    The data can be accessed as in a dictionary:
+    
+    >>> data = obsid.Obsid(3675508564)
+    >>> data['exptime']
+    <Quantity 8. s>
+    >>> data['linelist']
+    'Flare linelist 1'
+    >>> data['raster_fov']
+    '31.35x120'
+    >>> data['raster_step']
+    0.33
+    """
+
+    def __init__(self, obsid):
+        self.obsid = obsid
+        self.field_keys = {'Large linelist': 'linelist',
```

Comment:
Now this object is dict-like, and the keys can be got by doing ```Obsid.keys()```, I don't think there's any need to store these as an attribute.  Why not just define it as a local variable ```field_keys``` inside ```read_obsid```?